### ISSUE TITLE
Can't ifdef around a load directive

### ISSUE THREAD

**Issue by mastoj** [fable-compiler/Fable#416]
### Description

To be able to do "DEV" builds I would like to control the load of a file and some code using the `#if` directive.

Having this fsx-file:

```
#if DEV
#load "./MyModule.fs"
#endif

#if DEV
open MyModule
printfn "Hello A: %A" {b = "yolo"}
#endif
printfn "Done"
```

I would expect that to compile with or without `--symbols DEV`. If I provide the `DEV` symbol I expect a different output.

Note that the symbols is working on the second part, it is only around `#load` it doesn't work.

My `MyModule.fs` file looks like this:

```
module MyModule
type A = {b: string}
```

I have verified that it works fine when running `fsharpi` on mono.

**Comment by mastoj** [comment]
Maybe it is nesting of directives that isn't handled in general in fable?

**Comment by alfonsogarciacaro** [comment]
The directives are resolved by the F# compiler and are not visible in the AST that's provided to Fable. Is it possible to do this with normal F# scripts?

**Comment by mastoj** [comment]
It is possible with normal scripts, that's why expected it to work with fable.

**Comment by alfonsogarciacaro** [comment]
Hmm, interesting. As commented above, actually Fable doesn't do anything with directives (be it `#load` or `#if`), everything is handled by the F# compiler. I'll try to check when I've more time but may be the issue needs to be reported at the FCS repo.

**Comment by mastoj** [comment]
Maybe, but I think you are in a better position to tell if it should be
reported to fsc :).

It would make a nice feature for dev dependencies and code if it was
working.

**Comment by alfonsogarciacaro** [comment]
Here's where Fable extracts the project options (which will then by passed to the F# compiler) from a script file: https://github.com/fable-compiler/Fable/blob/e994005ecfb31cda70e3f571bf4961859220f259/src/fable/Fable.Client.Node/Main.fs#L164

There's no way to pass a compilation constant (a _symbol_) to `GetProjectOptionsFromScript` so I guess this is way the #load directive is not read at this point.

Maybe we could ask for a way to pass symbols to `GetProjectOptionsFromScript`?

**Comment by mastoj** [comment]
If you say so :)

**Comment by alfonsogarciacaro** [comment]
Closed by PR #429
